src/iccread.rs

```diff
@@ -1083,6 +1536,25 @@ impl Profile {
         profile
     }
 
+    pub fn new_cicp(cp: ColourPrimaries, tc: TransferCharacteristics) -> Option<Box<Profile>> {
```

Comment:
This will need C bindings. What will they look like?

Reply:
This would be simpler if qcms used cbindgen, but we can do the same thing by hand. In c_bindings.rs:
```rust
#[no_mangle]
pub unsafe extern "C" fn qcms_profile_create_cicp(
    cp: ColourPrimaries,
    tc: TransferCharacteristics,
) -> *mut Profile {
    …
}
```
And in qcms.h:
```C
typedef enum {
    // same values as iccread.rs
} qcms_ColourPrimaries;

typedef enum {
    // same values as iccread.rs
} qcms_TransferCharacteristics;

qcms_profile* qcms_profile_create_cicp(qcms_ColourPrimaries cp,
                                       qcms_TransferCharacteristics tc);
```

Then we'd just need to add `#[repr(C)]` to the `ColourPrimaries` and `TransferCharacteristics` enums in iccread.rs, and that should work.

However, I'm not sure I'll need to add this as a C binding. I think I'll be able to do all the qcms work from the rust side of mp4parse, so we may not need to add this at all.

Reply:
Ah, if you're able to call it from Rust then don't worry about the c bindings.